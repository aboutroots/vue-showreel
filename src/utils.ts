export const sleep = async (ms: number): Promise<void> => {
  /**
   * Sleep for given number of miliseconds
   */
  // eslint-disable-next-line no-promise-executor-return
  await new Promise((r) => setTimeout(r, ms));
};
