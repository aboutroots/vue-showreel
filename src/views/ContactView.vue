<template>
  <section class="ContactView container">
    <div class="contact-view-wrapper">
      <form class="contact-form" @submit="onSubmit">
        <h1>Contact us!</h1>
        <p>Amazing things are waiting for you, this is not a scam!!! 100% sure!</p>
        <input
          v-model="name"
          type="text"
          class="inp"
          required
          placeholder="Your name"
          aria-label="your name"
        />
        <input
          v-model="email"
          type="email"
          class="inp"
          required
          placeholder="Your email"
          aria-label="your email"
        />
        <input
          v-model="body"
          type="textarea"
          class="inp big"
          required
          placeholder="Your message"
          aria-label="your message"
        />
        <button class="btn btn-primary" type="submit">Submit</button>
      </form>
      <img src="@/assets/img/contract.svg" class="bg-img" alt="Background image with letter" />
    </div>
  </section>
</template>

<script lang="ts">
import Vue from 'vue';

interface State {
  name: string;
  email: string;
  body: string;
}
export default Vue.extend({
  name: 'ContactView',
  data() {
    return {
      name: '',
      email: '',
      body: '',
    } as State;
  },
  methods: {
    onSubmit(e: Event): void {
      e.preventDefault();
      const data = {
        name: this.name,
        email: this.email,
        body: this.body,
      };
      console.log('Submitted:', data);
      this.$toast.open('Thanks!');
    },
  },
});
</script>

<style lang="scss" scoped>
@import '@/assets/styles/theme.scss';

.contact-view-wrapper {
  display: flex;
  flex-wrap: wrap;
  align-items: center;
  margin: 0;
  @include tabletUp {
    margin: 0.5rem;
  }
}
.contact-form {
  display: flex;
  flex-direction: column;
  padding: 1rem;
  background-color: $dark2;
  border-radius: 12px;

  h1,
  p {
    text-align: center;
  }
  @include tabletUp {
    padding: 2rem;
  }
}
</style>
