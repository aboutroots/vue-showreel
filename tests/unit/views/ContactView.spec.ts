import ContactView from '@/views/ContactView.vue';
import { shallowMount } from '../utils';

describe('ContactView tests suite', () => {
  test('renders the component successfully', () => {
    const wrapper = shallowMount(ContactView);

    const container = wrapper.find('.ContactView');

    expect(container.exists()).toBeTruthy();
  });

  test('matches snapshot', () => {
    const wrapper = shallowMount(ContactView);
    expect(wrapper.element).toMatchSnapshot();
  });
});
