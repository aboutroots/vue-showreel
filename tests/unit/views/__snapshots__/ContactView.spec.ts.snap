// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContactView tests suite matches snapshot 1`] = `
<section
  class="ContactView container"
>
  <div
    class="contact-view-wrapper"
  >
    <form
      class="contact-form"
    >
      <h1>
        Contact us!
      </h1>
       
      <p>
        Amazing things are waiting for you, this is not a scam!!! 100% sure!
      </p>
       
      <input
        aria-label="your name"
        class="inp"
        placeholder="Your name"
        required="required"
        type="text"
      />
       
      <input
        aria-label="your email"
        class="inp"
        placeholder="Your email"
        required="required"
        type="email"
      />
       
      <input
        aria-label="your message"
        class="inp big"
        placeholder="Your message"
        required="required"
        type="textarea"
      />
       
      <button
        class="btn btn-primary"
        type="submit"
      >
        Submit
      </button>
    </form>
     
    <img
      alt="Background image with letter"
      class="bg-img"
      src="@/assets/img/contract.svg"
    />
  </div>
</section>
`;
